### PR TITLE
feat : Add LambdaTest cloud provider for real devices

### PR DESCRIPTION
**Pull request by anurag-lambdatest**
Add LambdaTest as a cloud provider for real devices

### CODE REVIEW

File: assets/locales/de/translation.json [hunk 1]
```diff
@@ -113,6 +113,7 @@
   "testingbotCredentialsRequired": "TestingBot key and secret are required!",
   "Perfecto SecurityToken is required": "Perfecto SecurityToken is required",
   "browserstackCredentialsRequired": "Browserstack username and access key are required!",
+  "lambdatestCredentialsRequired": "Lambdatest username and access key are required!",
```
Comment (KazuCocoa) [inline assets/locales/de/translation.json:116]: Could you add change only in english file? (`/en`)
https://github.com/appium/appium-inspector#localization

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "appium-inspector",
-  "version": "2022.2.1",
+  "version": "2022.2.2",
```
Comment (KazuCocoa) [inline package.json:3]: Could you exclude this change?

File: app/renderer/components/Session/AdvancedServerParams.js [hunk 3]
```diff
@@ -9,19 +9,21 @@ export default class AdvancedServerParams extends Component {
...
             <Panel header={t('Advanced Settings')}>
               <Row>
+                {
+                  serverType !== "lambdatest" && 
                 <Col span={6}>
                   <FormItem>
                     <Checkbox checked={!!server.advanced.allowUnauthorized} onChange={(e) => setServerParam('allowUnauthorized', e.target.checked, 'advanced')}>{t('allowUnauthorizedCerts')}</Checkbox>
                   </FormItem>
-                </Col>
+                </Col> }
```
Comment (KazuCocoa) [inline app/renderer/components/Session/AdvancedServerParams.js:26]: Could you adjust the indentation?

File: README.md [hunk 4]
```diff
@@ -25,7 +25,7 @@ Both apps have the exact same set of features, so you might find that simply ope
 
 ### Installing on macOS
 
-If you're using the desktop app on macOS, when you run it you will likely be greeted with some error about the app not being able to be opened, or not verified by Apple, or something similar. The easiest way to get around this is to run `xattr -cr` on the file you downloaded. So let's say you downloaded `appium-inspector.dmg`. Then you would run `xattr -cr appium-inspector.dmg` before opening the disk image. The same goes for the zip version (or the .app itself).
+If you're using the desktop app on macOS, when you run it you will likely be greeted with some error about the app not being able to be opened, or not verified by Apple, or something similar. The easiest way to get around this is to run `xattr -cr` on the file you downloaded. So let's say you downloaded `appium-inspector.dmg` and copy `Appium Inspector.app` in `/Applications` inside the disk image. Then you would run `xattr -cr "/Applications/Appium Inspector.app"` before opening it. The same goes for the zip version (or the .app itself).
```
Comment (KazuCocoa) [inline README.md:28]: Perhaps do these lines need update? These changes are already in main branch.

**Review by KazuCocoa — COMMENTED** [review]
Thank you!

- Could you sign properly?
- Could you exclude the package-lock.json at this time?

I've added some comments in lines as well.

File: app/renderer/components/Session/AdvancedServerParams.js (hunk 3 above)
Comment (anurag-lambdatest) [reply app/renderer/components/Session/AdvancedServerParams.js]: this is done in the latest commit, thank you .

File: package.json (hunk 2 above)
Comment (anurag-lambdatest) [reply package.json]: this is done in the latest commit, thank you .

File: assets/locales/de/translation.json (hunk 1 above)
Comment (anurag-lambdatest) [reply assets/locales/de/translation.json]: this is done in the latest commit, thank you .

File: README.md (hunk 4 above)
Comment (anurag-lambdatest) [reply README.md:28]: this is done in the latest commit, thank you .

**Comment by KazuCocoa** [comment]
Could you exclude `package-lock.json`'s deletion change?

File: app/renderer/actions/Session.js [hunk 5]
```diff
@@ -310,6 +310,38 @@ export function newSession (caps, attachSessId = null) {
...
+          desiredCapabilities['lt:options'].isRealMobile = true;
+          if (isProxyChecked) {
+            if(session.server.advanced.proxy == undefined) desiredCapabilities['lt:options'].proxyUrl = '';
+            else desiredCapabilities['lt:options'].proxyUrl = `${session.server.advanced.proxy}`;
+          }
+        } else {
+          desiredCapabilities['lambdatest:source'] = 'appiumdesktop';
+          desiredCapabilities['lambdatest:isRealMobile'] = true;
+          if (isProxyChecked) {
+            if(session.server.advanced.proxy == undefined) desiredCapabilities['lambdatest:proxyUrl'] = '';
```
Comment (KazuCocoa) [inline app/renderer/actions/Session.js:330]: `==` should be `===` in JS

File: app/renderer/actions/Session.js [hunk 6]
```diff
@@ -310,6 +310,38 @@ export function newSession (caps, attachSessId = null) {
...
+        host = session.server.lambdatest.hostname = process.env.LAMBDATEST_HOST || 'mobile-hub.lambdatest.com';
+        port = session.server.lambdatest.port = process.env.LAMBDATEST_PORT || 443;
+        path = session.server.lambdatest.path = '/wd/hub';
+        const isProxyChecked = session.server.advanced.useProxy;
+        username = session.server.lambdatest.username || process.env.LAMBDATEST_USERNAME;
+        if (desiredCapabilities.hasOwnProperty('lt:options')) {
+          desiredCapabilities['lt:options'].source = 'appiumdesktop';
+          desiredCapabilities['lt:options'].isRealMobile = true;
+          if (isProxyChecked) {
+            if(session.server.advanced.proxy == undefined) desiredCapabilities['lt:options'].proxyUrl = '';
```
Comment (KazuCocoa) [inline app/renderer/actions/Session.js:323]: `==` should be `===` in JS

**Review by KazuCocoa — COMMENTED** [review]
The CI runs `eslint app test` command. Could you make sure your changes have no lint error?
The package-lock still has diff. Please not include changes in there as this PR. Simply, you can copy current main branch's `package-lock` and paste it to this branch, and commit it, if you could not revert the changes.

File: app/renderer/actions/Session.js (hunk 5 above)
Comment (anurag-lambdatest) [reply app/renderer/actions/Session.js]: this is weird, we did copy the package-lock from main branch and even I can not find any difference in the file.

File: app/renderer/actions/Session.js [hunk 7]
```diff
@@ -310,6 +311,44 @@ export function newSession (caps, attachSessId = null) {
...
+      case ServerTypes.lambdatest:
+        host = session.server.lambdatest.hostname = process.env.LAMBDATEST_HOST || 'mobile-hub.lambdatest.com';
+        port = session.server.lambdatest.port = process.env.LAMBDATEST_PORT || 443;
+        path = session.server.lambdatest.path = '/wd/hub';
+        isProxyChecked = session.server.advanced.useProxy;
+        username = session.server.lambdatest.username || process.env.LAMBDATEST_USERNAME;
+        if (desiredCapabilities.hasOwnProperty.call(desiredCapabilities, 'lt:options')) {
+          desiredCapabilities['lt:options'].source = 'appiumdesktop';
+          desiredCapabilities['lt:options'].isRealMobile = true;
+          if (isProxyChecked) {
```
Comment (KazuCocoa) [inline app/renderer/actions/Session.js:323]: Can refer to `session.server.advanced.useProxy` directly here and below. Then, `isProxyChecked` could remove as only this block refers to it.

File: app/renderer/actions/Session.js (hunk 5 above)
Comment (KazuCocoa) [reply app/renderer/actions/Session.js]: Then, could you create a new PR with the necessary changes?
I guess GitHub diff messed by some reason.

File: app/renderer/actions/Session.js [hunk 8]
```diff
@@ -310,6 +311,44 @@ export function newSession (caps, attachSessId = null) {
...
         https = session.server.browserstack.ssl = (parseInt(port, 10) === 443);
         break;
+      case ServerTypes.lambdatest:
+        host = session.server.lambdatest.hostname = process.env.LAMBDATEST_HOST || 'mobile-hub.lambdatest.com';
+        port = session.server.lambdatest.port = process.env.LAMBDATEST_PORT || 443;
+        path = session.server.lambdatest.path = '/wd/hub';
+        isProxyChecked = session.server.advanced.useProxy;
+        username = session.server.lambdatest.username || process.env.LAMBDATEST_USERNAME;
+        if (desiredCapabilities.hasOwnProperty.call(desiredCapabilities, 'lt:options')) {
+          desiredCapabilities['lt:options'].source = 'appiumdesktop';
```
Comment (jlipps) [inline app/renderer/actions/Session.js:321]: all of this block is duplicated code, can you try to refactor it a bit?

Comment (rohithmone) [reply app/renderer/actions/Session.js:319]: Made some changes now. Please verify it

**Comment by jlipps** [comment]
Before I review again, please fix conflicts

**Comment by KazuCocoa** [comment]
Perhaps you could create a new branch base on this repository's main branch. Then, you may need to cherry-pick necessary changes from this PR to the branch. and create a new PR.
The new one should not have package-lock, readme, and newlines only changes in translation files.

**Comment by anurag-lambdatest** [comment]
closing the current PR, will be raising a new one.

**Pull request closed without merging**